pyproject.toml

```diff
@@ -43,7 +44,7 @@ numpy = [
     { version = "^1.24.2", optional = true, python = ">=3.9,<3.12" },
     { version = "^1.26.2", optional = true, python = ">=3.12" },
 ]
-pyarrow = { version = "^14", optional = true }
+pyarrow = { version = ">=14,<19", optional = true }
```

Comment:
For the SDK I think it's best if we don't add a upper limit for our dependencies unless we have a good reason to.

Reply:
This was just mainly to keep the current type of behaviour but I agree we could use this instead.

```toml
pyarrow = { version = ">=14", optional = true }
```